### PR TITLE
ctpg: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **ctpg/1.3.5**

ctpg is "Compile Time Parser Generator" library.
It is header only and written in C++17.

Currently it doesn't supports Visual Studio which failed to compile with MSB60006.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
ctpg seems not to suppport clang.
While I'm asking about it, I made this PR draft.

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/ctpg/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "1.3.2":
```
Comment (madebr) [inline recipes/ctpg/all/conandata.yml:2]: 1.3.4 is out now, which should support clang

File: recipes/ctpg/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import os
+from conans import ConanFile, CMake, tools
+
+class DawHeaderLibrariesTestConan(ConanFile):
```
Comment (madebr) [inline recipes/ctpg/all/test_package/conanfile.py:4]: Copy past detected!
Copyright lawyers will soon arrive at your door!
```suggestion
from conans import ConanFile, CMake, tools


class TestPackageConan(ConanFile):
```

Comment (toge) [reply recipes/ctpg/all/test_package/conanfile.py:4]: Sorry my mistake.

File: recipes/ctpg/all/conandata.yml (hunk 1 above)
Comment (toge) [reply recipes/ctpg/all/conandata.yml]: Thank you for your information!
I update this PR.

**Comment by toge** [comment]
In my local environment, I was able to successfully compile ctpg with clang 10.
Something's wrong...

**Comment by madebr** [comment]
I can reproduce with the `conanio/clang10` docker image.
My system clang 12 compiler builds ctpg without issue.

Perhpaps you need to bump the minimum required clang version.
https://github.com/peter-winter/ctpg/issues/15 does not mention a version.

**Comment by toge** [comment]
Thank you for valuable information!
I can reproduce with `conan/clang10` too.
But I can't reproduce with my personal environment (Ubuntu 11.1).

Those version is slightly different.
```
Ubuntu clang version 10.0.1-++20211003084855+ef32c611aa21-1~exp1~20211003085243.2
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

```
clang version 10.0.0 
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/bin
```

After all, I checked ctpg is only compiled only with clang12 and clang13.
I will bump the minimum requied clang version as you wrote.

Thank you.

**Review by madebr — DISMISSED** [review]

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**